API/Authentication.md

```diff
@@ -0,0 +1,95 @@
+
+### Authorization header
+
+HTTP supports sending a standard `Authorization` header in requests. The header consists of a type (indicating what kind of a credential it is) and a value separated by whitespace. An example header can look like this: `Basic aHR0cHM6Ly95b3V0dS5iZS9kUXc0dzlXZ1hjUQ==`.
+
+You can find the list of types [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#authentication_schemes). The one most commonly seen in APIs is the so-called `Bearer` token. As explained in the [Bearer Token Usage RFC](https://datatracker.ietf.org/doc/html/rfc6750#section-1.2), a bearer token is a:
+> A security token with the property that any party in possession of the token (a "bearer") can use the token in any way that any other party in possession of it can.
+When the server generates a token, it should make it opaque (meaning they can't be interpreted by anyone other than the server).
+
+`Authorization` header is the preferred mechanism if your API will be used in platforms other than the browser. It's easier to control than cookies outside of browsers, and it doesn't come with security concerns around using query strings (described below).
+
+### Basic auth
+
+Basic auth is the simplest authentication method. It's a type of `Authorization` header authentication, where the "type" value is `Basic`.
+
+### Query string
+
+Query string is a part of the URL used to provide additional parameters. Query string starts after question mark character (`?`) and delimits multiple parameters with amspersands (`&`). Each parameter consists of a name and a string value separated by an equal sign (`=`).
+
+For example, URL `example.com?api_key=123&token=abc` has two query params: `api_key` with value `123`, and `token` with value `abc`.
+
+Query string can be used to transmit keys and tokens in requests to the server, similar to cookies and `Authorization` header. At a first glance, this might seem insecure, but if it's sent over the `https` protocol, query string is encrypted with the rest of the request and it's visible only by the client and the server.
+
+However, there are other concerns when using query strings to transmit sensitive data:
+- URLs (including query strings) are visible in server logs (although, you can [filter parameters](https://guides.rubyonrails.org/action_controller_overview.html#parameters-filtering) like these in Rails)
+- URLs are saved in browser history (when used in navigation)
+- you have to include the key/token in every request, but normally you want to reserve the query string for other parameters (like filters, pagination, sorting, etc.)
+
+Consider using query strings for authentication only when other methods are unavailable.
+
+## Methods
+
+### API keys
+
+API keys are unique strings used to identify clients.
```

Comment:
Think of API keys like normal passwords and you should treat them like passwords